velox/functions/prestosql/GreatestLeast.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <functions/Macros.h>
+#include <cmath>
+
+namespace facebook::velox::functions::details {
+
+template <typename TExec, typename T, bool isLeast>
+struct ExtremeValueFunction;
+
+template <typename TExec, typename T>
+using LeastFunction = ExtremeValueFunction<TExec, T, true>;
+
+template <typename TExec, typename T>
+using GreatestFunction = ExtremeValueFunction<TExec, T, false>;
+
+/**
+ * This class implements two functions:
+ *
+ * greatest(value1, value2, ..., valueN) → [same as input]
+ * Returns the largest of the provided values.
+ *
+ * least(value1, value2, ..., valueN) → [same as input]
+ * Returns the smallest of the provided values.
+ *
+ * For DOUBLE and REAL type, NaN is considered as the biggest according to
+ * https://github.com/prestodb/presto/wiki/Presto-NaN-behavior
+ **/
+template <typename TExec, typename T, bool isLeast>
+struct ExtremeValueFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<T>& result,
+      const arg_type<T>& firstElement,
+      const arg_type<Variadic<T>>& remainingElement) {
```

Comment:
This function has default null behavior, hence, call will not be called if any argument is NULL.

I wonder if arg_type here can be replaced with null_free_arg_type. Even if not, VELOX_USER_CHECK below can be removed.

Reply:
It seems like using `null_free_arg_type` will cause compilation error.
error: static assertion failed: UDF must implement at least one of `call`, `callNullable`, or `callNullFree` functions.
This error happens also if the output and input types of the functions do not match the ones used in registration.

So for this fix I just simply deleted `VELOX_USER_CHECK`

Reply:
Fixed